tidb/src/tidb/sql.clj

```diff
@@ -246,36 +247,13 @@
 (defmacro with-txn
   "Executes body in a transaction, with a timeout, automatically retrying
   conflicts and handling common errors."
-  [op [c conn opts] & body]
+  [op opts & body]
```

Comment:
What is the connection management mechanism here, not quite understand, will we meet the connection problem like before like doing rollback?

Reply:
No, I think. Previously a connection may be shared and accessed by two thread: the thread of worker (where we call `attach-txn-info` in `invoke!`, it's outside of `with-txn`) and the thread spawned by `timeout` (within `with-txn`), and now we just try to obtain txn info and attach it to `op` inside of the `timeout` thread.